maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/GoalRenderer.java

```diff
@@ -0,0 +1,543 @@
+        }
+        endTable();
+        endSection();
+    }
+
+    private void renderTableCellWithCode(String text) {
+        renderTableCellWithCode(text, Optional.empty());
+    }
+
+    private void renderTableCellWithCode(String text, URI link) {
+        renderTableCellWithCode(text, Optional.of(link));
+    }
+
+    private void renderTableCellWithCode(String text, Optional<URI> link) {
+        sink.tableCell();
+        sink.inline(Semantics.CODE);
+        if (link.isPresent()) {
+            sink.link(link.get().toString(), null);
+        }
+        sink.text(text);
+        if (link.isPresent()) {
+            sink.link_();
+        }
+        sink.inline_();
+        sink.tableCell_();
+    }
+
+    private void renderParameterOverviewTableRow(Parameter parameter) {
+        sink.tableRow();
+        // name
+        try {
+            renderTableCellWithCode(
+                    format("goal.parameter.name", parameter.getName()), new URI(null, null, parameter.getName()));
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Cannot create fragment link from " + parameter.getName(), e);
+        }
+
+        // type
+        Map.Entry<String, Optional<URI>> type = getLinkedType(parameter, true);
+        renderTableCellWithCode(type.getKey(), type.getValue());
+
+        // since
+        String since = StringUtils.defaultIfEmpty(parameter.getSince(), "-");
+        renderTableCellWithCode(since);
+
+        // description
+        sink.tableCell();
+        String description;
+        String context = "Parameter " + parameter.getName() + " in goal " + descriptor.getGoal();
+        if (StringUtils.isNotEmpty(parameter.getDeprecated())) {
+            String deprecated = getXhtmlWithValidatedLinks(parameter.getDescription(), context);
+            description = format("goal.parameter.deprecated", deprecated);
+        } else if (StringUtils.isNotEmpty(parameter.getDescription())) {
+            description = getXhtmlWithValidatedLinks(parameter.getDescription(), context);
+        } else {
+            description = getI18nString("goal.nodescription");
+        }
+        sink.rawText(description);
+        renderTableCellDetail("goal.parameter.defaultValue", parameter.getDefaultValue());
+        renderTableCellDetail("goal.parameter.property", getPropertyFromExpression(parameter.getExpression()));
+        renderTableCellDetail("goal.parameter.alias", parameter.getAlias());
+        sink.tableCell_();
+
+        sink.tableRow_();
+    }
+
+    private void renderParameterDetails(Iterator<Parameter> parameters) {
+
+        startSection(getI18nString("goal.parameter.details"));
+
+        while (parameters.hasNext()) {
+            Parameter parameter = parameters.next();
+            // TODO: rework parameter, currently 3 are generated: one from skin, one from startSection (all lowercase) and one manually
+            sink.anchor(parameter.getName());
```

Comment:
Which anchor should we keep for backwards compatibility reasons?

Reply:
I need to think about this and look at the output first.

Reply:
For as long as you don't link the anchor or need it stable (language-agnostic), there is no reason to invoke `Sink#anchor()` directly. I would drop it. The anchors from the skin are a skin problem because we don't know anything about them at this stage. So dropping 307 and 310 seems right to me.

Reply:
I link the anchor from the same page. Although for this use case I could use the new format created via HtmlTools.encodeId(...) called from `AbstractMavenReportRenderer.startSection(...)` potentially there exist some links leveraging those anchors externally. So we have IMHO three options:

1. Keep the anchor backwards compatible (but maybe invalid as it was done with xdoc in https://github.com/apache/maven-plugin-tools/blob/b7138b0a7744e31d43c2c16894203cd6debc1514/maven-plugin-tools-generators/src/main/java/org/apache/maven/tools/plugin/generator/PluginXdocGenerator.java#L649, i.e. case-sensitive names) or
2. Switch to the new format leveraging `HtmlTools.encodeId(...)` potentially breaking existing external deeplinks.
3. Create both formats old and new and use new one internally

The anchor being created by the skin is orthogonal as it is impossible to predict its value.

Reply:
OK, I see. Let's do the following:
* Support both: old and new and mark the old one in code to be removed with Doxia 2 move
* Remove old approach when plugin code is rebased to Doxia 2.0.0

The latter won't happen before end of year.

> The anchor being created by the skin is orthogonal as it is impossible to predict its value.

Correct, therefore I wouldn't even put it into consideration.